contracts/core/OriginCore.sol

```diff
@@ -412,4 +474,120 @@ contract OriginCore is OriginCoreInterface, OriginCoreConfig {
 
         return kernelHash;
     }
+    /**
+     * @notice private method to check vote validity.
+     *
+     * @dev This method will revert the transaction if validator is not
+     *       eligible for voting.
+     *
+     * @param _voteHash The hash of the transition part of the meta-block
+     *                    header at the source block.
+     * @param _v V of the signature.
+     * @param _r R of the signature.
+     * @param _s S of the signature.
+     *
+     * @return signer_ Address of validator who has signed the vote.
+     * @return weight_ Weight of validator at given meta-block height.
+     */
+    function checkVoteValidity(
+        bytes32 _voteHash,
+        uint8 _v,
+        bytes32 _r,
+        bytes32 _s
+
+    )
+        private
+        view
+        returns (address signer_, uint256 weight_)
+    {
+        // As per https://github.com/ethereum/go-ethereum/pull/2940
+        bytes memory prefix = "\x19Ethereum Signed Message:\n32";
+        _voteHash = keccak256(
+            abi.encodePacked(
+                prefix,
+                _voteHash
+            )
+        );
+
+        signer_ = ecrecover(
+            _voteHash,
+            _v,
+            _r,
+            _s
+        );
+
+        /* Header of last meta block. */
+        MetaBlock.Header storage latestMetaBlockHeader = reportedHeaders[head];
+        /* Current meta-block height. */
+        uint256 currentHeight = latestMetaBlockHeader.kernel.height.add(1);
+        weight_ = stake.weight(currentHeight, signer_);
+
+        require(
+            weight_ != 0,
+            'Only validator with non zero weight can vote.'
+        );
+    }
+
+    /**
+     * @notice private method to save vote into the seal.
+     *
+     * @dev This method assumes that vote has been successfully verified. It
+     *      will revert the transaction if validator already voted for this
+     *      transition object.
+     *
+     * @param _transition The hash of the transition part of the meta-block
+     *                    header at the source block.
+     * @param _signer Address of validator who has signed vote object.
+     * @param _weight Weight of validator at given meta-block height.
+     */
+    function saveVote(
+        bytes32 _transition,
+        address _signer,
+        uint256 _weight
+    )
+        private
+        returns(uint256 totalVoteWeight_)
+    {
+        MetaBlock.Seal storage sealObj = seal[_transition];
```

Comment:
Why `sealObj` and not just `seal`?

Reply:
Because of solidity warning of shadowing storage variable named `seal`. 

But we can change this to `seal` now because the storage variable is now `seals`.

Reply:
👍